### PR TITLE
docs: ui capabilities switch PTL-1579

### PR DESCRIPTION
**Pull request by Gareth-Spencer-Genesis**
ATENTION! YOU SHOULD NOW BRANCH FROM PREPROD WHEN YOU UPDATE 

  - Please check the [internal contributions guide](https://www.notion.so/genesisglobal/Contributing-new-documentation-75953fb245f246ff872789035451a0c4) for details
  
__________

Thank you for contributing to the documentation.

Do the changes you have made apply to both Current and Previous versions?
No

Have you done a trial build to check all new or changed links?
Yes

Is there anything else you would like us to know?
No

**This week's exciting advice from the style guide**

- Write your headings in sentence case:

  - A good example
    This is a correct heading.
  - A Bad Example
    This is not a correct heading.

### CODE REVIEW

File: docs/001_develop/03_client-capabilities/008_interaction/switch.mdx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+Declaration
+```html
+<rapid-switch></rapid-switch>
+```
+Usage
+```typescript
+export function MyComponent() {
+  const [switchValue, setSwitchState] = useState(false);
+  const handleSwitchChange = (event) => {
+    setSwitchState(!!event.target.checked);
```
Comment (patrickoneill-genesis) [inline docs/001_develop/03_client-capabilities/008_interaction/switch.mdx:67]: is this just resetting the switch state?

File: docs/001_develop/03_client-capabilities/008_interaction/switch.mdx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+```typescript
+import { Component, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+@Component({
+  selector: 'my-root',
+  template: `
+    <rapid-switch
+      name="isSelected"
+      [checked]="isSelected"
+      (change)="onSwitchChange($event)"
```
Comment (patrickoneill-genesis) [inline docs/001_develop/03_client-capabilities/008_interaction/switch.mdx:94]: same here -  we don't have it in the foundation example

**Review by patrickoneill-genesis — COMMENTED** [review]
Looks good - just wondering why we are passing the value into the switch in the angular/react examples and not the foundation?

It should work without resetting the value via templates.

File: docs/001_develop/03_client-capabilities/008_interaction/switch.mdx (hunk 2 above)
Comment (Gareth-Spencer-Genesis) [reply docs/001_develop/03_client-capabilities/008_interaction/switch.mdx:98]: OK I added to the genesis code example.

File: docs/001_develop/03_client-capabilities/008_interaction/switch.mdx (hunk 1 above)
Comment (Gareth-Spencer-Genesis) [reply docs/001_develop/03_client-capabilities/008_interaction/switch.mdx:72]: This is a copy and paste from the checkbox matt did. Im not a react developer, but I think he is manually tracking the state of the checkbox. Feel free to update the code to correct it if you want to

Comment (patrickoneill-genesis) [reply docs/001_develop/03_client-capabilities/008_interaction/switch.mdx:72]: https://chatgpt.com/share/672b7fdc-07b4-8009-af8c-73644c6291f1 - checkout this chat gpt link (i'm years out of date with react too)

I created a blank app in create for react and angular to test these things. I understand it's a pain.

If it speeds up the process - mark the section as needs testing so we know to check it later

File: examples/ui/client-capabilities/interaction/switch.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { CodeLabel, CodeSection } from '../../documentationBase';
+import { registerComponents } from '../../rapidImports';
+
+registerComponents();
```
Comment (patrickoneill-genesis) [inline examples/ui/client-capabilities/interaction/switch.js:4]: Fix this import

File: docs/001_develop/03_client-capabilities/008_interaction/switch.mdx [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+<rapid-switch></rapid-switch>
+```
+
+Usage
+```typescript
+@customElement({
+  name: 'my-element',
+  template: html`
+    <rapid-switch
+      :checked="${x => x.isSelected}"
```
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/008_interaction/switch.mdx:43]:
```suggestion
      ?checked="${x => x.isSelected}"
```

File: docs/001_develop/03_client-capabilities/008_interaction/switch.mdx [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+  <thead><tr><th>Name</th><th>Type</th><th>Description</th><th>Example</th></tr></thead>
+  <tbody>
+    <tr>
+      <td>checked</td>
+      <td><code>boolean</code></td>
+      <td>Sets or retrieves the current state of the component. **Default: `false`**</td>
+<td>
+
+```typescript
+<rapid-switch checked>
```
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/008_interaction/switch.mdx:129]:
```suggestion
<rapid-switch ?checked=${sync(x => x.isSelected, 'boolean'}>
```

File: docs/001_develop/03_client-capabilities/008_interaction/switch.mdx [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+| label   | The label text.|
+| checked-indicator | The checked indicator.|
+| checked-message | The text label when the switch is selected.|
+| switch  | The input control.|
+| unchecked-message | The text label when the switch is not selected.|
+
+### Events fired
+| Name    | Description                                                |
+|---------|------------------------------------------------------------|
+| change  | Fires the event when the radio component changes its state |
```
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/008_interaction/switch.mdx:192]: Let's put an example of using the change event here, [like this](https://github.com/genesiscommunitysuccess/docs/pull/1895/files#diff-de5bad40a087c6287b3752c04167e66df977b97ed8150a9d0566e108c975053dR213-R236)

File: docs/001_develop/03_client-capabilities/008_interaction/switch.mdx [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+  <thead><tr><th>Name</th><th>Type</th><th>Description</th><th>Example</th></tr></thead>
+  <tbody>
+    <tr>
+      <td>checked</td>
+      <td><code>boolean</code></td>
+      <td>Sets or retrieves the current state of the component. **Default: `false`**</td>
+<td>
+
+```typescript
+<rapid-switch ?checked=${sync(x => x.isSelected, 'boolean'}>
```
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/008_interaction/switch.mdx:129]: My bad
```suggestion
<rapid-switch ?checked=${sync(x => x.isSelected, 'boolean')}>
```

**Review by matteematt — APPROVED** [review]

**Pull request merged**